src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -526,6 +526,34 @@ public AuthenticationConfiguration useCertificateCredential(SecurityFactory<X509
     }
 
     /**
+     * Create a new configuration which is the same as this configuration, but it has {@link org.wildfly.security.credential.store.CredentialStore}
+     * attached for referencing later.
+     *
+     * @param name of {@code CredentialStore}
+     * @param type of {@code CredentialStore}, {@code null} to use default
+     * @param attributes to initialize instance of {@code CredentialStore}
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration useCredentialStore(String name, String type, Map<String, String> attributes) {
+        Assert.checkNotNullParam("name", name);
```

Comment:
FYI we are in the middle of some other changes in AuthenticationClient. the build methods should now be 'final' this way we can be sure none of the individual implementations are affecting building.